main/SQLHelper.cpp

```diff
@@ -5176,6 +5194,100 @@ void CSQLHelper::UpdateUVLog()
 	}
 }
 
+bool CSQLHelper::UpdateCalendarMeter(const int HardwareID, const char* DeviceID, const unsigned char unit, const unsigned char devType, const unsigned char subType, bool dayCalendar, float counter, float usage, const char* date)
+{
```

Comment:
Maybe a small check if the 'date' field is correct, for instance it's length + dashes on the right possition